### DISCUSSION TITLE
[Question] Why not use AddressBindContext.ServerOptions instead of `listenOptions` in AddressBinder.BindAsync?

### DISCUSSION THREAD

**Discussion by wu-yafeng** [dotnet/aspnetcore#25033] [General] [answered]
see [AddressBinder.BindAsync(...)](https://github.com/dotnet/aspnetcore/blob/4995bbf109d8c16faa6cc3dd8e8104f7b5f72008/src/Servers/Kestrel/Core/src/Internal/AddressBinder.cs#L24)

It seems `listenOptions` are redundant,use `context.ServerOptions.ListenOptions.ToArray()` instead

``` csharp
var strategy = CreateStrategy(
                context.ServerOptions.ListenOptions.ToArray(),
                context.Addresses.ToArray(),
                context.ServerAddressesFeature.PreferHostingUrls);
```

**Comment by Tratcher** [comment]
@halter73

**Comment by halter73** [comment, marked as answer]
I think it is technically redundant [given how it's currently used](https://github.com/dotnet/aspnetcore/blob/dc4670e8d5e8026d3ce832a1e7b5205d34913bce/src/Servers/Kestrel/Core/src/KestrelServer.cs#L282), but it's an internal API. Having the listenOptions as a separate parameter does make  AddressBinder.BindAsync more flexible even if nothing is taking advantage of that flexibility.

**Comment by wu-yafeng** [comment]
Thanks!
